doc/xml/contributing.xml

```diff
@@ -321,87 +393,324 @@
+
+            <p><b>Debugging with files</b></p>
+
+            <p>Sometimes it is useful to look at files that were generated during the test. The default for running any test is that, at the start/end of the test, the test harness will clean up all files and directories created. To override this behavior, a single test run must be specified and the option <code>--no-cleanup</code> provided. Again, continuing with the check command, from <b>define.yaml</b> above, there are four tests. Below, test one will be run and nothing will be cleaned up so that the files and directories in <path>test/test-0</path> can be inspected.</p>
+
+            <code-block>
+pgbackrest/test/test.pl --vm-out --dev --module=command --test=check --run=1 --no-cleanup
+            </code-block>
+        </section>
+    </section>
+
+    <section id="option">
+        <title>Adding an Option</title>
+
+        <p>Options can be added to a command or multiple commands. Options can be configuration file only, command-line only or valid for both. Once an option is successfully added, <file>config.auto.*</file>, <file>define.auto.*</file> and <file>parse.auto.*</file> files will automatically be generated by the build system.</p>
+
+        <p>To add an option, two files need be to be modified:</p>
+        <list>
+            <list-item><file>src/build/config/config.yaml</file></list-item>
+            <list-item><file>doc/xml/reference.xml</file></list-item>
+        </list>
+
+        <p>These files are discussed in the following sections along with how to verify the <code>help</code> command output.</p>
+
+        <section id="config-file">
+            <title>config.yaml</title>
+
+            <p>There are detailed comment blocks above each section that explain the rules for defining commands and options. Regarding options, there are two types: 1) command line only, and 2) configuration file. With the exception of secrets, all configuration file options can be passed on the command line. To configure an option for the configuration file, the <id>section:</id> key must be present.</p>
+
+            <p>The <id>option:</id> section is broken into sub-sections by a simple comment divider (e.g. <code># Repository options</code>) under which the options are organized alphabetically by option name. To better explain this section, two hypothetical examples will be discussed. For more details, see <link url="{[github-url-src]}/build/config/config.yaml">config.yaml</link>.</p>
+
+            <section id="option-ex1">
+                <title>Example 1: hypothetical command line only option </title>
+
+                <code-block>
+  set:
+    type: string
+    command:
+      backup:
+        depend:
+          option: stanza
+        required: false
+      restore:
+        default: latest
```

Comment:
The default command role has been renamed to main in PR #1410. So if we commit that first we can add command role here, or leave it in the config.yaml documentation.